### PR TITLE
[Prim][PIR] silu prim sink

### PR DESCRIPTION
**Pull request by kevincheng2**
### PR types

New features
### PR changes

Others 
### Description

1. prim `silu` op sink c++ 
2. add `is_half_dtype()` function in `utils.h`
3. modify `mean`、`silu`、`sqrt`、`batch_norm`、`softmax`、`layer_norm` operator dtype judgment method

### CODE REVIEW

File: paddle/fluid/primitive/composite/composite.h [hunk 1]
```diff
@@ -203,6 +203,29 @@ Tensor softmax_decomp(const Tensor& x, const int& axis) {
...
+      org_dtype == phi::DataType::FLOAT16 || org_dtype == phi::DataType::UINT16;
+  if (need_cast) {
+    x_tmp = cast<T>(x, phi::DataType::FLOAT32);
+  }
+
+  // res = x / (1 + exp(-x))
+  auto one = full<T>(phi::vectorize(x.dims()), 1, x_tmp.dtype());
+  auto exp_temp =
+      exp<T>(full<T>(phi::vectorize(x.dims()), -1, x_tmp.dtype()) * x_tmp);
+  auto res = divide<T>(x_tmp, exp_temp + one);
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]: suggest symbol overload

File: paddle/fluid/primitive/composite/composite.h [hunk 2]
```diff
@@ -203,6 +203,29 @@ Tensor softmax_decomp(const Tensor& x, const int& axis) {
   }
 }
 
+template <typename T>
+Tensor silu_decomp(const Tensor& x) {
+  auto org_dtype = x.dtype();
+  auto x_tmp = x;
+
+  bool need_cast =
+      org_dtype == phi::DataType::FLOAT16 || org_dtype == phi::DataType::UINT16;
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]: suggest provide common utils like is_half_dtype(...)

Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]: why unint16 instead of BFLOAT16?

**Review by cyber-pioneer — APPROVED** [review]
LGTM
